cmake/IgnUtils.cmake

```diff
@@ -1804,3 +1804,455 @@ macro(_ign_cmake_parse_arguments prefix options oneValueArgs multiValueArgs)
+      "export IGN_GAZEBO_SYSTEM_PLUGIN_PATH=@CMAKE_INSTALL_PREFIX@/${plugin_path}\n")
+    endforeach()
+    foreach(gui_plugin_path ${gui_plugins_path})
+      file(APPEND ${CMAKE_CURRENT_SOURCE_DIR}/setup.sh.in 
+      "export IGN_GUI_PLUGIN_PATH=@CMAKE_INSTALL_PREFIX@/${gui_plugin_path}\n")
+    endforeach()
+    foreach(variable_export_path ${variable_export_paths})
+      string(REPLACE "," "=" variable_export_path ${variable_export_path})
+      file(APPEND ${CMAKE_CURRENT_SOURCE_DIR}/setup.sh.in
+      "export ${variable_export_path}\n")
+    endforeach()
+    file(WRITE ${CMAKE_CURRENT_SOURCE_DIR}/export_path.sh.in "#!/usr/bin/env bash\nsource @CMAKE_INSTALL_PREFIX@/share/@PROJECT_NAME@/setup.sh\n")
+    
+    configure_file(
+      "setup.sh.in"
+      "${CMAKE_CURRENT_BINARY_DIR}/share/setup.sh" @ONLY
+    )
+
+    install(FILES
+      ${CMAKE_CURRENT_BINARY_DIR}/share/setup.sh
+      DESTINATION share/${PROJECT_NAME}
+    )
+    
+    configure_file(
+      "export_path.sh.in"
+      "${CMAKE_CURRENT_BINARY_DIR}/share/export_path" @ONLY
+    )
+
+    install(PROGRAMS
+      ${CMAKE_CURRENT_BINARY_DIR}/share/export_path
+      DESTINATION bin
+    )
+
+  else()
+    message(FATAL_ERROR
+    "Unknown export type")
+  endif()
+  
+
+endmacro()
+#####################################################
+# ign_export_variable(variable_name variable_path)
+# Exports path of mentioned varaible.
+# Descriptions of the function arguments are as follows:
+#
+# <variable_name>: Name of the variable whoose path is to be exported.
+#
+# <variable_path>: Path of the above mentioned variable.
+#
+macro(ign_export_variable variable_name variable_path)
+  list(APPEND variable_export_paths "${variable_name},${variable_path}")
+endmacro()
+
+#######################################################
+# ign_add_plugins(path_to_plugins)
+# Installs all the plugins inside the folder using various arguments from the user
+# Also provides the path to ign_environment_hook()
+# Other than common link_libraries and directories,you can simply add specific plugin dependencies one by one after this macro
+# The deafult file type is .cc for plugins,you can add other file types using PLUGIN_EXTENSION argument
+# For adding GUI plugins,one can simply pass GUI true as argument
+# Descriptions of the function arguments are as follows:
+#
+# <path_to_plugin>: Path of the folder where plugins are present
+#
+# [GUI]: Optional.If provided, it will build the plugin as a gui plugin.
+#
+# [INSTALL_DESTINATION]: Optional. Destination of the folder where plugins will be installed.
+#                        Defaults to lib for normal plugins and lib/gui for gui plugins.
+#
+# [SCOPE]: Optional. STATIC, SHARED, or MODULE may be given to specify the type of library to be created.
+#          Defaults to SHARED library
+#
+# [PLUGIN_EXTENSION]: Optional. Macros will assume files of this extension inside the specifc folder as plugins.
```

Comment:
I've elaborated many times why this is a bad idea. I know it _feels_ nice but Im still (very strongly) against it. It saves the user literally 2 characters. If you want to add this it should be added in your repo not in `ignition-cmake`

Reply:
I have removed default extension